classes/Group.php

```diff
@@ -106,18 +106,18 @@ public function __construct($id = null, $idLang = null, $idShop = null)
      * @since   1.0.0
      * @version 1.0.0 Initial version
      */
-    public static function getGroups($idLang, $idShop = false)
+    public static function getGroups($idLang, $allShops = false)
     {
         $shopCriteria = '';
-        if ($idShop) {
+        if (!$allShops) {
```

Comment:
We can't just change semantics of input parameter. 

Call `Groups::getGroups($lang)` should return all groups regardless of shop association. That's current semantics, and it's obvious from the code that this is intended.

With your change, this call would return groups associated only with current shop. That's a big compatibility issue

I recommend to change the fix to follow original semantics.

$idShop == false --> return all groups
$idShop == true --> return groups associated with **current** shop (from context)
$idShop == number --> return groups associated with this **specific** shop

Reply:
I've reverted my changes, except the added missing use of $shopCriteria.
I've also changed the if, to add the missing behavior, if a number is given through the parameter.
Should now work as you mentioned above.